src/Disks/createVolume.cpp

```diff
@@ -1,17 +1,57 @@
 #include "createVolume.h"
 
+#include <Disks/SingleDiskVolume.h>
+#include <Disks/VolumeJBOD.h>
+#include <Disks/VolumeRAID1.h>
+
+#include <boost/algorithm/string.hpp>
+
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_RAID_TYPE;
+}
+
 VolumePtr createVolumeFromReservation(const ReservationPtr & reservation, VolumePtr other_volume)
```

Comment:
`other_volume`? What is it? Seems like we use the only name and max_size from it?

Reply:
Personally I don't like whole idea of mixing volume descriptions and working volumes. Here, working volume is being constructed from volume description.